gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/ExecutionVisitor.java

```diff
@@ -128,6 +140,15 @@ public EntityResponse visit(DataFetcherNode dataFetcherNode) {
             entitiesRequest.getEntityType(),
             executionContext.getTimestampAttributeId());
 
+    Filter entitiesRequestFilter =
+        childFilter != null
```

Comment:
When result is empty - it seems that the method `constructFilterFromChildNodesResult` returns `Filter.getDefaultInstance()`
Is the defaultInstance equals `null`?

Reply:
> Is the defaultInstance equals null?

No. It's just an empty filter instance